### PR TITLE
Move `fallback_has_occurred` state tracking to `FnCtxt`

### PR DESCRIPTION
**Pull request by compiler-errors**
Removes a ton of callsites that defaulted to `false`

### CODE REVIEW

File: compiler/rustc_hir_typeck/src/fallback.rs [hunk 1]
```diff
@@ -9,14 +9,14 @@ use rustc_middle::ty::{self, Ty};
 impl<'tcx> FnCtxt<'_, 'tcx> {
     /// Performs type inference fallback, returning true if any fallback
     /// occurs.
```
Comment (deltragon) [inline compiler/rustc_hir_typeck/src/fallback.rs:11]: This comment is now outdated, same with `fallback_if_possible` below.

Comment (compiler-errors) [reply compiler/rustc_hir_typeck/src/fallback.rs:11]: Thanks :smiley:

**Comment by eholk** [comment]
r+

**Pull request merged**